core/src/main/java/org/jdbi/v3/core/Jdbi.java

```diff
@@ -327,8 +329,15 @@ public Handle open() {
      * @throws X any exception thrown by the callback
      */
     public <R, X extends Exception> R withHandle(HandleCallback<R, X> callback) throws X {
+        if (threadHandle.get() != null) {
```

Comment:
I'm slightly worried some user will have code that expects to be able to nest `withHandle` to get two separate database connections.

Reply:
I agree that this could be seen as breaking behavior by some people. There is actually a test that depended on this assumption, and had to be modified under the new behavior.

With that said, I think this is a step forward. I've seen multiple users who were confused by the existing behavior of opening separate handles per invocation. Even some of my coworkers.

Reply:
Yeah, my coworker was similarly confused.  I'm convinced.